service/service.go

```diff
@@ -50,13 +50,34 @@ var getMongoDataStore = func(ctx context.Context, cfg *config.Configuration) (da
 }
 
 // getKafkaProducer creates a new Kafka Producer
-var getKafkaProducer = func(ctx context.Context, kafkaBrokers []string, topic string, envMax int, kafkaVersion string) (kafka.IProducer, error) {
+var getKafkaProducer = func(ctx context.Context, cfg *config.Configuration, topic string) (kafka.IProducer, error) {
 	producerChannels := kafka.CreateProducerChannels()
 	pConfig := &kafka.ProducerConfig{
-		KafkaVersion:    &kafkaVersion,
-		MaxMessageBytes: &envMax,
+		KafkaVersion:    &cfg.KafkaVersion,
+		MaxMessageBytes: &cfg.KafkaMaxBytes,
 	}
-	return kafka.NewProducer(ctx, kafkaBrokers, topic, producerChannels, pConfig)
+
+	if cfg.KafkaSecProtocol == config.KafkaSecProtocolTLS {
+		pConfig.SecurityConfig = kafka.GetSecurityConfig(
+			cfg.KafkaSecCACerts,
+			cfg.KafkaSecClientCert,
+			cfg.KafkaSecClientKey,
+			cfg.KafkaSecSkipVerify,
```

Comment:
Assuming the security protocol is set to TLS should we also validate these fields are not null/empty?

For example an error on start up stating `"KafkaSecClientKey expected but was nil"` is clear and easy to understand/diagnose and saves having to dig into logs & configs to work out what is wrong.

Reply:
have addressed this